docs/source/python/integration/extending.rst

```diff
@@ -50,12 +50,23 @@ This will not include other parts of the Arrow API, which you will need
 to include yourself (for example ``arrow/api.h``).
 
 When building C extensions that use the Arrow C++ libraries, you must add
-appropriate linker flags. We have provided functions ``pyarrow.get_libraries``
-and ``pyarrow.get_library_dirs`` which return a list of library names and
+appropriate linker flags. We have provided functions ``pa.get_libraries``
+and ``pa.get_library_dirs`` which return a list of library names and
 likely library install locations (if you installed pyarrow with pip or
 conda). These must be included when declaring your C extensions with
 setuptools (see below).
 
+.. note::
+
+   The PyArrow-specific C++ code is now a part of the PyArrow source tree
+   and not Arrow C++. That means the header files and arrow_python library
+   are not necessarily installed in the same location as that of Arrow.
```

Comment:
Should we add something like ".. and will no longer be automatically findable by CMake"? 

We no longer install an ArrowPython cmake config file in `/lib/cmake`, so my understanding is that `find_package(arrow_python)` in CMake therefore won't work anymore? 

cc @kou @pitrou

Reply:
Ah, right.

Can we suggest our C data/stream interface as an alternative of `arrow/python/api.h`?

Reply:
I suppose we could suggest that as an alternative depending on your use cases, but there are more things the pyarrow C++ API let you do than just converting arrays / tables between python and C++, so it's not a full replacement.

And you still have the Arrow C++ dependency, so I am not sure if avoiding the pyarrow C++ dependency gives then that much of benefit? (regarding building/packaging complexity)

Reply:
It's not my use case. I was thinking about this case: https://issues.apache.org/jira/browse/ARROW-16340?focusedCommentId=17612958#comment-17612958

Reply:
I think we should still support the wrap/unwrap functions [as documented](https://arrow.apache.org/docs/python/integration/extending.html).

Reply:
Would it work if I added something similar to:
".. and will no longer be automatically findable by CMake. For a subset of functionality available in PyArrow C++ API one can use C [data](https://arrow.apache.org/docs/dev/format/CDataInterface.html)/[stream](https://arrow.apache.org/docs/dev/format/CStreamInterface.html) interface as an alternative."

Reply:
IMHO there is not point in advertising the C Data Interface as an alternative, since the PyArrow C++ API is still there.